src/Compilers/CSharp/Test/Symbol/Symbols/GenericConstraintTests.cs

```diff
@@ -6862,5 +6862,190 @@ static void Main(string[] args)
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(14, 9)
+            );
+        }
+
+        [Fact]
+        [WorkItem(41779, "https://github.com/dotnet/roslyn/issues/41779")]
+        public void Bug41779_NonGeneric()
+        {
+            var source =
+@"interface I
+{
+    object GetService();
+}
+
+static class Program
+{
+    static void GetService(this I obj){}
+    
+    static void M(I provider)
+    {
+        provider.GetService<>();
+        provider.GetService<>().ToString();
+        provider.GetService<>();
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (12,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(12, 9),
+                // (12,18): error CS0308: The non-generic method 'I.GetService()' cannot be used with type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_HasNoTypeVars, "GetService<>").WithArguments("I.GetService()", "method").WithLocation(12, 18),
+                // (13,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>().ToString();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(13, 9),
+                // (13,18): error CS0308: The non-generic method 'I.GetService()' cannot be used with type arguments
+                //         provider.GetService<>().ToString();
+                Diagnostic(ErrorCode.ERR_HasNoTypeVars, "GetService<>").WithArguments("I.GetService()", "method").WithLocation(13, 18),
+                // (14,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(14, 9),
+                // (14,18): error CS0308: The non-generic method 'I.GetService()' cannot be used with type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_HasNoTypeVars, "GetService<>").WithArguments("I.GetService()", "method").WithLocation(14, 18)
+            );
+        }
+
+
+        [Fact]
+        [WorkItem(41779, "https://github.com/dotnet/roslyn/issues/41779")]
+        public void Bug41779_Class()
+        {
+            var source =
+@"using System;
+class C
+{
+    String GetService(){ return String.Empty; }
```

Comment:
>String GetService(){ return String.Empty; } [](start = 4, length = 43)

Is presence of this API necessary? Do we even bind to it in this scenario?

Reply:
> I'm fairly certain we do, since removing this line acts as the difference between correct errors and does not emit in SharpLab on master at the moment.

Does the presence of the extension method below make a difference?

---
In reply to: [442352158](https://github.com/dotnet/roslyn/pull/45033#discussion_r442352158) [](ancestors = 442352158)